.github/workflows/build-test-coverage.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Test & Coverage
+on:
+    pull_request:
+    push:
+
+jobs:
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.12"
+    - name: Run tests
+      run: |
+        pip install -q -U pip
+        pip install -q pipenv
+        pipenv install --dev
+        pipenv run pytest --cov-report=xml --cov=newrelic_lambda_cli tests
+    - name: Upload coverage report
+      run: bash -c "$(curl -s https://codecov.io/bash)"
```

Comment:
Can we use the codecov action instead of executing arbitrary downloaded bash here? Seems better for security.

You'll want something like this, may have to specify specific files if it doesn't pick them up by default.

```
      - name: Upload Coverage to Codecov
        uses: codecov/codecov-action@v3
          fail_ci_if_error: true
```

Reply:
Addressed in the revised PR